chainer/utils/precision.py

```diff
@@ -0,0 +1,34 @@
+import numpy
+import functools
+
+
+def mixed_precision(fn):
+    """Decorator to perform forward computation in FP32 for FP16 inputs,
+       returning outputs casted back to FP16. Do nothing for FP32 and FP64
+       inputs.
+    """
+    @functools.wraps(fn)
+    def wrapper(self, in_data):
+        flag = any([x.dtype == numpy.float16 for x in in_data])
```

Comment:
Why `any` ?

Reply:
I think of cases `float16` data and that of other dtype, e.g. integer for labels, are supplied in `in_data`.

Reply:
I got your point. Makes sense.

Reply:
Fixed!